File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Server/Restore.cs
using System;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Operations.Backups;
using Raven.Client.ServerWide;
using Raven.Client.ServerWide.Operations;

namespace Raven.Documentation.Samples.ClientApi.Operations.Server
{
    public class Restore
    {
        private class Foo
        {
            public class DeleteDatabasesOperation
            {
                /*
                #region restore_1
                public RestoreBackupOperation(RestoreBackupConfiguration restoreConfiguration)
                #endregion
                */

                #region restore_2
                public class RestoreBackupConfiguration
                {
                    public string DatabaseName { get; set; }

                    public string BackupLocation { get; set; }

                    public string LastFileNameToRestore { get; set; }

                    public string DataDirectory { get; set; }

                    public string EncryptionKey { get; set; }
                }
                #endregion
            }
        }

        public Restore()
        {
            using (var store = new DocumentStore())
            {
                #region restore_3
                RestoreBackupConfiguration config = new RestoreBackupConfiguration()
                {
                    BackupLocation = @"C:\backups\Northwind",
                    DatabaseName = "Northwind"
                };
                RestoreBackupOperation restoreOperation = new RestoreBackupOperation(config);
                store.Maintenance.Server.Send(restoreOperation)
                    .WaitForCompletion();
                #endregion
            }
        }
    }
}

File: Documentation/5.1/Samples/nodejs/client-api/operations/server/Restore.js
import {DocumentStore, RestoreBackupOperation} from 'ravendb';
import {IServerOperation} from "ravendb";
import {OperationIdResult} from "ravendb";

let restoreConfiguration;
//document_store_creation
const store = new DocumentStore(["http://localhost:8080"], "Northwind2");
store.initialize();
//const session = store.openSession();
{
    //region restore_1
    const restoreBackupOperation = new RestoreBackupOperation(restoreConfiguration, "nodeTag");
    //endregion
}

//region restore_2
export interface RestoreBackupConfigurationBase {
    databaseName,
    lastFileNameToRestore,
    dataDirectory,
    encryptionKey,
    disableOngoingTasks,
    skipIndexes,
    type,
    backupEncryptionSettings
}
//endregion
{
//region restore_3
    restoreConfiguration = {
        databaseName: "Northwind",
        skipIndexes: false
    }
    const restoreBackupOperation = RestoreBackupOperation(restoreConfiguration, "A");
    const restoreResult = await store.maintenance.server.send(restoreBackupOperation);
//endregion
}